File: Notes/app/src/main/java/com/serhii/apps/notes/Notes.java
package com.serhii.apps.notes;

import android.app.Application;

import com.serhii.apps.notes.common.AppConstants;
import com.serhii.apps.notes.control.managers.InactivityManager;
import com.serhii.core.CoreEngine;
import com.serhii.core.log.Log;
import com.serhii.core.log.LogImpl;
import com.serhii.core.security.Hash;
import com.serhii.core.security.impl.crypto.Result;

public class Notes extends Application {

    private static final String TAG = "Notes";

    @Override
    public void onCreate() {
        Log.info(TAG, "onCreate(), IN");
        super.onCreate();
        // Initialize application configuration
        Log.setTag(AppConstants.APP_LOG_TAG);
        // Enable detailed logs
        if (BuildConfig.DEBUG) {
            Log.info(TAG, "onCreate(), running debug build");
            Log.setDetailedLogs(true);
        }
        // Init inactivity manager
        InactivityManager.getInstance().initManager(this);
        Log.info(TAG, "onCreate(), OUT");
    }
}

File: Notes/app/src/main/java/com/serhii/apps/notes/Notes.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.apps.notes

import android.app.Application
import com.serhii.apps.notes.common.App
import com.serhii.apps.notes.control.preferences.PreferenceManager
import com.serhii.core.FileSystem
import com.serhii.core.log.Log

/**
 * Performs setup of app components
 */
class Notes : Application() {

    override fun onCreate() {
        super.onCreate()
        // Setup log settings
        Log.tag = App.APP_LOG_TAG
        val enabled = PreferenceManager.detailedLogsEnabled(applicationContext)
        Log.enableDetailedLogs(enabled)
        Log.setVersionCode(BuildConfig.VERSION_NAME)
        // Setup file system settings
        FileSystem.setFilePath(this)
    }
}